x-pack/legacy/plugins/siem/server/lib/events/elasticsearch_adapter.ts

```diff
@@ -280,14 +280,14 @@ const getDataFromHits = (sources: EventSource, category?: string, path?: string)
     const item: EventSource = get(source, sources);
     if (Array.isArray(item) || isString(item) || isNumber(item)) {
       const field = path ? `${path}.${source}` : source;
-      category = field.split('.')[0];
-      if (isEmpty(category) && baseCategoryFields.includes(category)) {
-        category = 'base';
+      let fieldCategory = field.split('.')[0];
```

Comment:
So as you probably noticed we try to keep our usage of `let` down to a minimum and very low instances of it in the code base. 9 times of 10 you can clean up a `let` variable with a `const` by introducing a small pure function which returns the value it is mutating.

Then you get the extra advantage of being able to add a unit test case for that new decomposed smaller function and the larger function which gives more safety.

In this example it would be something like this (review and write tests for anything I recommend ;-):

```ts
const getFieldCategory = (field: string): string => {
  const fieldCategory = field.split('.')[0];
  if (isEmpty(fieldCategory) && baseCategoryFields.includes(fieldCategory)) {
    return 'base';
  } else {
    return field.split('.')[0];
  }
};
``` 

And then you would get the fieldCategory like so:

```ts
...
      const field = path ? `${path}.${source}` : source;
      const fieldCategory = getFieldCategory(field);
      return [
        ...accumulator,
        {
          category: fieldCategory,
          field,
          values: item,
          originalValue: item,
        } as DetailItem,
      ];
...
```

Reply:
Sounds like a plan! I had given thought to making it a function over using a `let` but didn't want to seem like I was being too heavy-handed in the refactor. I'll give that a go then :)